packages/plugin-page-view-tracking/src/plugin-campaign-tracker.ts

```diff
@@ -0,0 +1,53 @@
+import { AttributionBrowserOptions, Campaign } from '@amplitude/analytics-types';
+import { CampaignTracker } from '@amplitude/analytics-client-common';
+import { BASE_CAMPAIGN } from './constant';
+import { Storage } from '@amplitude/analytics-types';
+
+export class PluginCampaignTracker {
+  private _currentCampaign: Campaign = {
+    ...BASE_CAMPAIGN,
+  };
+  private _isNewCampaign = true;
+  ready: Promise<unknown>;
+  campaignTracker: CampaignTracker;
+
+  constructor(apiKey: string, storage: Storage<Campaign>, options: AttributionBrowserOptions) {
+    this.campaignTracker = new CampaignTracker(apiKey, {
+      ...options,
+      trackPageViews: false,
+      track: /* istanbul ignore next */ () => Promise.resolve(),
+      onNewCampaign: /* istanbul ignore next */ () => () => undefined,
+      storage,
+    });
+    this.ready = new Promise((resolve) => resolve(this.refreshCampaignState()));
```

Comment:
i think this is an example of why we should not write this as a class since no way to await on constructor

Reply:
Okay, let me try to make it in a function.